docs: rust (#1816)
pypa/cibuildwheel: docs/faq.md

[Homebrew]: https://brew.sh/
[delocate]: https://github.com/matthew-brett/delocate

### Building Rust wheels

If you build Rust wheels, you need to download the Rust compilers in manylinux.

> If you support 32-bit windows, you need to add this as a potential target. You

Author: ```suggestion
If you support 32-bit Windows, you need to add this as a potential target. You
```

> can do this on GitHub Actiions, for example, with:

Reviewer: ```suggestion
can do this on GitHub Actions, for example, with:
```

> ```toml

Author: ```suggestion
```yaml
```

CIBW_BEFORE_ALL_LINUX: curl -sSf https://sh.rustup.rs | sh -s -- -y
CIBW_BEFORE_ALL_WINDOWS: rustup target add i686-pc-windows-msvc
CIBW_ENVIRONMENT_LINUX: "PATH=$HOME/.cargo/bin:$PATH"
```

> You probably need to skip PyPy (if using PyO3, anyway), and Rust provides cargo for musllinux 32-bit:

Author: ```suggestion
You probably need to skip PyPy (if using PyO3, anyway), and Rust does not provide Cargo for musllinux 32-bit:
```

> You probably need to skip PyPy (if using PyO3, anyway), and Rust does not provide Cargo for musllinux 32-bit:
> 
> ```toml
> [tool.cibuildwheel]
> skip = ["pp*", "*-musllinux_i686"]

Author: ```suggestion
Rust does not provide Cargo for musllinux 32-bit, so that needs to be skipped:

```toml
[tool.cibuildwheel]
skip = ["*-musllinux_i686"]
```

(Not sure if there's an issue or if it was just me)

```

> Also see [maturin-action](https://github.com/PyO3/maturin-action) which is optimized for Rust wheels and can cross-compile (and can build 32-bit musl, for example).

Author: ```suggestion
Also see [maturin-action](https://github.com/PyO3/maturin-action) which is optimized for Rust wheels, builds the non-Python Rust modules once, and can cross-compile (and can build 32-bit musl, for example).
```
